client/src/main/java/org/apache/rocketmq/client/impl/producer/TopicPublishInfo.java

```diff
@@ -29,6 +29,15 @@
     private List<MessageQueue> messageQueueList = new ArrayList<MessageQueue>();
     private volatile ThreadLocalIndex sendWhichQueue = new ThreadLocalIndex();
     private TopicRouteData topicRouteData;
+    private long notFoundFlag = -1;
```

Comment:
Is it more appropriate to name it as `lastNotFoundTimestamp`?

Reply:
I think u are right.

Have done on the thrid commit.